File: GXpert/GXpert.Web/Modules/ServerTypes/Content/ContentCategoryRow.ts
﻿import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib";

export interface ContentCategoryRow {
    Id?: number;
    Name?: string;
    Description?: string;
    IsGlobal?: boolean;
    MediaCount?: number;
    Thumbnail?: string;
    ListingThumbnail?: string;
    IsActive?: number;
    InsertUserId?: number;
    InsertDate?: string;
    UpdateUserId?: number;
    UpdateDate?: string;
}

export abstract class ContentCategoryRow {
    static readonly idProperty = 'Id';
    static readonly nameProperty = 'Name';
    static readonly localTextPrefix = 'Content.ContentCategory';
    static readonly lookupKey = 'Content.ContentCategory';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<ContentCategoryRow>('Content.ContentCategory') }
    static async getLookupAsync() { return getLookupAsync<ContentCategoryRow>('Content.ContentCategory') }

    static readonly deletePermission = 'Content:ContentManagement:View';
    static readonly insertPermission = 'Content:ContentManagement:View';
    static readonly readPermission = 'Content:ContentManagement:View';
    static readonly updatePermission = 'Content:ContentManagement:View';

    static readonly Fields = fieldsProxy<ContentCategoryRow>();
}

File: GXpert/GXpert.Web/Modules/Content/ContentCategory/ContentCategoryRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using Serenity.Extensions.Entities;
using System;
using System.ComponentModel;

namespace GXpert.Content;

[ConnectionKey("Default"), Module("Content"), TableName("ContentCategories")]
[DisplayName("Content Category"), InstanceName("Content Category")]
[ReadPermission(PermissionKeys.ContentManagement.View)]
[ModifyPermission(PermissionKeys.ContentManagement.View)]
[ServiceLookupPermission("Administration:General")]
[LookupScript("Content.ContentCategory")]
public sealed class ContentCategoryRow : LoggingRow<ContentCategoryRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Id"), Identity, IdProperty]
    public int? Id { get => fields.Id[this]; set => fields.Id[this] = value; }

    [DisplayName("Name"), Size(100), NotNull, QuickSearch, NameProperty]
    public string Name { get => fields.Name[this]; set => fields.Name[this] = value; }

    [DisplayName("Description"), Size(1000)]
    public string Description { get => fields.Description[this]; set => fields.Description[this] = value; }

    [DisplayName("Is Global"), NotNull]
    public bool? IsGlobal { get => fields.IsGlobal[this]; set => fields.IsGlobal[this] = value; }

    [DisplayName("Media Count"), NotNull]
    public int? MediaCount { get => fields.MediaCount[this]; set => fields.MediaCount[this] = value; }

    [DisplayName("Thumbnail"), Size(500)]
    public string Thumbnail { get => fields.Thumbnail[this]; set => fields.Thumbnail[this] = value; }

    [DisplayName("Listing Thumbnail"), Size(500)]
    public string ListingThumbnail { get => fields.ListingThumbnail[this]; set => fields.ListingThumbnail[this] = value; }

    [DisplayName("Is Active"), NotNull,DefaultValue(1)]
    public short? IsActive { get => fields.IsActive[this]; set => fields.IsActive[this] = value; }

    public class RowFields : LoggingRowFields
    {
        public Int32Field Id;
        public StringField Name;
        public StringField Description;
        public BooleanField IsGlobal;
        public Int32Field MediaCount;
        public StringField Thumbnail;
        public StringField ListingThumbnail;
        public Int16Field IsActive;

    }
}